### ISSUE TITLE
[BUG]  cli_demo.py 使用 Qwen-14B-Chat-Int4，出现RuntimeError: "addmm_impl_cpu_" not implemented for 'Half'

### ISSUE THREAD

**Issue by dage0127** [QwenLM/Qwen#611]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

错误：RuntimeError: "addmm_impl_cpu_" not implemented for 'Half'
模型：Qwen-14B-Chat-Int4
代码：cli_demo.py 
环境： python 3.9
硬件： GPU 

/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/cuda/__init__.py:138: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11000). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() > 0
Using `disable_exllama` is deprecated and will be removed in version 4.37. Use `use_exllama` instead and specify the version with `exllama_config`.The value of `use_exllama` will be overwritten by `disable_exllama` passed in `GPTQConfig` or stored in your config file.
Warning: please make sure that you are using the latest codes and checkpoints, especially if you used Qwen-7B before 09.25.2023.请使用最新模型和代码，尤其如果你在9月25日前已经开始使用Qwen-7B，千万注意不要使用错误代码和模型。
Try importing flash-attention for faster inference...
Warning: import flash_attn rotary fail, please install FlashAttention rotary to get higher efficiency https://github.com/Dao-AILab/flash-attention/tree/main/csrc/rotary
Warning: import flash_attn rms_norm fail, please install FlashAttention layer_norm to get higher efficiency https://github.com/Dao-AILab/flash-attention/tree/main/csrc/layer_norm
Warning: import flash_attn fail, please install FlashAttention to get higher efficiency https://github.com/Dao-AILab/flash-attention
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00,  5.36it/s]
Welcome to use Qwen-Chat model, type text to start chat, type :h to show command help.
(欢迎使用 Qwen-Chat 模型，输入内容即可进行对话，:h 显示命令帮助。)

Note: This demo is governed by the original license of Qwen.
We strongly advise users not to knowingly generate or allow others to knowingly generate harmful content, including hate speech, violence, pornography, deception, etc.
(注：本演示受Qwen的许可协议限制。我们强烈建议，用户不应传播及不应允许他人传播以下内容，包括但不限于仇恨言论、暴力、色情、欺诈相关的有害信息。)

User> [ERROR] Query is empty
User> [ERROR] Query is empty
User> [ERROR] Query is empty
User> 你好
Traceback (most recent call last):
  File "/home/haitaiwork/gptllm/gen/Qwen/cli_demo.py", line 212, in <module>
    main()
  File "/home/haitaiwork/gptllm/gen/Qwen/cli_demo.py", line 200, in main
    for response in model.chat_stream(tokenizer, query, history=history, generation_config=config):
  File "/home/haitaiwork/gptllm/.cache/huggingface/modules/transformers_modules/Qwen-14B-Chat-Int4/modeling_qwen.py", line 1273, in stream_generator
    for token in self.generate_stream(
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/utils/_contextlib.py", line 35, in generator_context
    response = gen.send(None)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/transformers_stream_generator/main.py", line 931, in sample_stream
    outputs = self(
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/haitaiwork/gptllm/.cache/huggingface/modules/transformers_modules/Qwen-14B-Chat-Int4/modeling_qwen.py", line 1108, in forward
    transformer_outputs = self.transformer(
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/haitaiwork/gptllm/.cache/huggingface/modules/transformers_modules/Qwen-14B-Chat-Int4/modeling_qwen.py", line 938, in forward
    outputs = block(
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/haitaiwork/gptllm/.cache/huggingface/modules/transformers_modules/Qwen-14B-Chat-Int4/modeling_qwen.py", line 639, in forward
    attn_outputs = self.attn(
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/haitaiwork/gptllm/.cache/huggingface/modules/transformers_modules/Qwen-14B-Chat-Int4/modeling_qwen.py", line 464, in forward
    mixed_x_layer = self.c_attn(hidden_states)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/auto_gptq/nn_modules/qlinear/qlinear_cuda_old.py", line 273, in forward
    out = torch.matmul(x, weight)
RuntimeError: "addmm_impl_cpu_" not implemented for 'Half'

### 期望行为 | Expected Behavior

运行正常

### 运行环境 | Environment

```Markdown
- OS:Ubuntu 18.04
- Python:3.9
- Transformers:4.35.0
- PyTorch:2.1.0
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):12.1
(llm) ~/gen/Qwen$ python -c 'import torch; print(torch.version.cuda)'
12.1
```

**Comment by jklj077** [comment]
根据错误提示，调用CUDA失败，实际在CPU执行了（量化模型依赖autogptq，autogptq<5.0不支持CPU执行）
```
/home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/cuda/init.py:138: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11000). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org/ to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
```

需要按您的实际情况，升级nvidia驱动或者使用低版本的CUDA

**Comment by LiuGuBiGu** [comment]
> 根据错误提示，调用CUDA失败，实际在CPU执行了（量化模型依赖autogptq，autogptq<5.0不支持CPU执行）
> 
> ```
> /home/haitaiwork/gptllm/anaconda3/envs/llm/lib/python3.9/site-packages/torch/cuda/init.py:138: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11000). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org/ to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
> ```
> 
> 需要按您的实际情况，升级nvidia驱动或者使用低版本的CUDA

请问下我用openai_api.py部署Qwen-7B-Chat-Int4，请求api的时候也报错"addmm_impl_cpu_" not implemented for 'Half'，AutoGPTQ版本是最新的0.6.0，最新的支持在CPU上运行了吗
